lib/src/generator/dartdoc_generator_backend.dart

```diff
@@ -44,14 +44,32 @@ class DartdocGeneratorBackendOptions implements TemplateOptions {
       this.useBaseHref = false});
 }
 
+class SidebarGenerator<T extends Documentable> {
+  final Template template;
+  final Map<T, String> _renderCache = {};
+
+  SidebarGenerator(this.template);
+
+  // Retrieve the render for a specific key, or generate it using the given
+  // template data if you need.
+  String getRenderFor(T key, TemplateData templateData) {
+    return _renderCache[key] ??= template.renderString(templateData);
+  }
+}
+
 /// Base GeneratorBackend for Dartdoc's supported formats.
 abstract class DartdocGeneratorBackend implements GeneratorBackend {
   final DartdocGeneratorBackendOptions options;
   final Templates templates;
+  final SidebarGenerator<Library> sidebarForLibrary;
+  final SidebarGenerator<Container> sidebarForContainer;
 
   DartdocGeneratorBackend(
       DartdocGeneratorBackendOptions options, this.templates)
-      : options = (options ?? DartdocGeneratorBackendOptions());
+      : options = (options ?? DartdocGeneratorBackendOptions()),
```

Comment:
These parens are unnecessary, right? What about `??=`?

Reply:
Parentheses are unnecessary, deleted.   `??=` however appears to be illegal in initializer lists.